### PR TITLE
[READY]Ports a shittone of abductor related PRs

### PR DESCRIPTION
**Pull request by ghost**
# Document the changes in your pull request

Abductors are very cool dudes, so why not to make them even cooler?

https://github.com/tgstation/tgstation/pull/63738
https://github.com/BeeStation/BeeStation-Hornet/pull/1308
https://github.com/BeeStation/BeeStation-Hornet/pull/1827
https://github.com/BeeStation/BeeStation-Hornet/pull/3234
https://github.com/BeeStation/BeeStation-Hornet/pull/959
https://github.com/BeeStation/BeeStation-Hornet/pull/1473
https://github.com/tgstation/tgstation/pull/48034
https://github.com/tgstation/tgstation/pull/45387
https://github.com/tgstation/tgstation/pull/63127

Also moves gland types into separated files

# Spriting

I am not able of making ingame screenshots with new sprites, check original PR's or icon bot.

# Changelog

:cl: PKPenguin321, Fury McFlurry,  Skoglol, SomeAngryMiner, AdamElTablawy, Bastian0930, carshalash, FernandoJ8, SuperSlayer
rscadd: A shrink ray for abductors that they can buy at their console for 2 experiment points. Anything can be shrunken, and shrunken things can be walked over/seen past (for example, you can shrink a wall and then walk around it). If a human is shrunk, they will drop all of their belongings, move slower, and take 2x damage.
imageadd: Shrink ray sprites by Fury McFlurry. Thank you!
admin: Admins can now pick solo abductor in traitor panel.
rscadd: Abductors now can't spawn first 8 minutes of the round.
imageadd: Bluespace navigation beacons now have a new fancy model, more visually pleasing and intuitive, with the bonus of also providing light.
rscadd: You can now by an agent ID as an abductor
tweak: the abductor chem dispenser now has a few extra chems in it
rscadd: New abductee objectives
rscadd: Adds descriptions to abductor glands upon examination
imageadd: Resprites all of the abductors surgical tools
/:cl:

### CODE REVIEW

**Comment by Marmio64** [comment]
i dont think abductors need their guns to be able to shoot ion and freeze beams, nor does the ship need a recharger. The pistol is supposed to be an emergency side-arm that you use very sparingly (if you do end up using it, find a station recharger). In addition they're supposed to be weak fighters in general, they have the ability to instantly teleport away back to their ship.

**Comment by ghost** [comment]
> i dont think abductors need their guns to be able to shoot ion and freeze beams, nor does the ship need a recharger. The pistol is supposed to be an emergency side-arm that you use very sparingly (if you do end up using it, find a station recharger). In addition they're supposed to be weak fighters in general, they have the ability to instantly teleport away back to their ship.

Ion's aren't kinda offensive and are supposed to protect you from nerds with energy weaponry, and freezing beams are also for defensive means, so I think it is OK.

**Comment by tattax** [comment]
we already have abductor UI also freezing beams and flash protection are meh since they can mass produce helmets + are already hard to counterplay if you don't have a tinfoil helmet

**Comment by ghost** [comment]
> we already have abductor UI also freezing beams and flash protection are meh since they can mass produce helmets + are already hard to counterplay if you don't have a tinfoil helmet

Getting fucked as an antag > getting fucked as a crewmember, because you get an antagonist... Well... Not very often, and as a crewmember you literally can play every round. Also abductors aren't very antagonistic, since they often can even buff people, so i don't think that buffing tham is a SO bad think. But perhaps i can exclude the abductor gun part from this PR.

File: yogstation.dme [hunk 1]
```diff
@@ -1372,6 +1372,21 @@
...
+#include "code\modules\antagonists\abductor\equipment\glands\heal.dm"
+#include "code\modules\antagonists\abductor\equipment\glands\mindshock.dm"
+#include "code\modules\antagonists\abductor\equipment\glands\plasma.dm"
+#include "code\modules\antagonists\abductor\equipment\glands\pop.dm"
+#include "code\modules\antagonists\abductor\equipment\glands\quantum.dm"
+#include "code\modules\antagonists\abductor\equipment\glands\shock.dm"
+#include "code\modules\antagonists\abductor\equipment\glands\slime.dm"
+#include "code\modules\antagonists\abductor\equipment\glands\spiderman.dm"
+#include "code\modules\antagonists\abductor\equipment\glands\trauma.dm"
+#include "code\modules\antagonists\abductor\equipment\glands\ventcrawl.dm"
```
Comment (ghost) [inline yogstation.dme]:
```suggestion
#include "code\modules\antagonists\abductor\equipment\glands\ventcrawling.dm"
```

**Comment by VaelophisNyx** [comment]
problem is the fan favorite of abductors is plasma vomit and lightning bolts; both of which cause severe issues for the round at hand.

**Comment by ghost** [comment]
> problem is the fan favorite of abductors is plasma vomit and lightning bolts; both of which cause severe issues for the round at hand.

I don't think that litterally all abductor players only think about how to fuck up the round, and also what is the problem to just remove their glands with surgery?

**Comment by VaelophisNyx** [comment]
> also what is the problem to just remove their glands with surgery?

Good luck removing a plasma vomit gland when your medbay has been reduced to slag because of a plasma fire. Or convulsing on the ground because you keep getting shocked by the patient involuntarily.

**Comment by ghost** [comment]
> Good luck removing a plasma vomit gland when your medbay has been reduced to slag because of a plasma fire. Or convulsing on the ground because you keep getting shocked by the patient involuntarily.

I can nerf it if needed

File: code/datums/components/shrink.dm [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/datum/component/shrink
+	var/olddens
+	var/oldopac
+	dupe_mode = COMPONENT_DUPE_HIGHLANDER
+
+/datum/component/shrink/Initialize(shrink_time)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	var/atom/parent_atom = parent
+	parent_atom.transform = parent_atom.transform.Scale(0.5,0.5)
```
Comment (ToasterBiome) [inline code/datums/components/shrink.dm:10]: Don't like this from the one simple fact that we can't trust players to NOT roleplay as smol beans

File: code/modules/antagonists/abductor/equipment/abduction_gear.dm [hunk 3]
```diff
@@ -717,6 +728,7 @@ Congratulations! You are now trained for invasive xenobiology research!"}
 	desc = "Abduct with style - spiky style. Prevents digital tracking."
 	icon_state = "alienhelmet"
 	item_state = "alienhelmet"
+	flash_protect = 1
```
Comment (ToasterBiome) [inline code/modules/antagonists/abductor/equipment/abduction_gear.dm]: No

File: code/modules/shuttle/spaceship_navigation_beacon.dm [hunk 4]
```diff
@@ -7,13 +7,15 @@
 /obj/machinery/spaceship_navigation_beacon
 	name = "Bluespace Navigation Gigabeacon"
 	desc = "A device that creates a bluespace anchor that allow ships jump near to it."
-	icon = 'icons/obj/abductor.dmi'
-	icon_state = "core"
+	icon = 'icons/obj/machines/NavBeacon.dmi'
```
Comment (ToasterBiome) [inline code/modules/shuttle/spaceship_navigation_beacon.dm:10]: unrelated change needs to be atomized

**Review by ToasterBiome — CHANGES_REQUESTED** [review]
Don't see the point in separating all the glands, but whatever. Needs to be tested.

File: code/modules/shuttle/spaceship_navigation_beacon.dm (hunk 4 above)
Comment (ghost) [reply code/modules/shuttle/spaceship_navigation_beacon.dm:10]: What do you mean

File: code/datums/components/shrink.dm (hunk 2 above)
Comment (ghost) [reply code/datums/components/shrink.dm:10]: It is a temporary effect, so it seems ok

File: code/modules/antagonists/abductor/equipment/abduction_gear.dm (hunk 3 above)
Comment (ghost) [inline code/modules/antagonists/abductor/equipment/abduction_gear.dm]:
```suggestion
```

File: code/modules/antagonists/abductor/equipment/abduction_gear.dm [hunk 5]
```diff
@@ -411,6 +411,17 @@ Congratulations! You are now trained for invasive xenobiology research!"}
...
+	name = "shrink ray blaster"
+	desc = "This is a piece of frightening alien tech that enhances the magnetic pull of atoms in a localized space to temporarily make an object shrink. \
+			That or it's just space magic. Either way, it shrinks stuff."
+	ammo_type = list(/obj/item/ammo_casing/energy/shrink)
+	item_state = "shrink_ray"
+	icon_state = "shrink_ray"
+	fire_delay = 30
+	selfcharge = 1//shot costs 200 energy, has a max capacity of 1000 for 5 shots. self charge returns 25 energy every couple ticks, so about 1 shot charged every 12~ seconds
+	trigger_guard = TRIGGER_GUARD_ALLOW_ALL// variable-size trigger, get it? (abductors need this to be set so the gun is usable for them)
+
```
Comment (ghost) [inline code/modules/antagonists/abductor/equipment/abduction_gear.dm]:
```suggestion
	pin = /obj/item/firing_pin/abductor
	
```

**Comment by tattax** [comment]
> Getting fucked as an antag > getting fucked as a crewmember, because you get an antagonist... Well... Not very often, and as a crewmember you literally can play every round. Also abductors aren't very antagonistic, since they often can even buff people, so i don't think that buffing tham is a SO bad think. But perhaps i can exclude the abductor gun part from this PR.

getting sunglasses ain't hard + it's just stunning htem with the flash they can still be recalled by the scientist

**Comment by ghost** [comment]
Ok

**Comment by JamieD1** [comment]
@SomeguyManperson

**Comment by SomeguyManperson** [comment]
Abductors do not need a "shrink ray" they arent meant to fuck with people on station

**Comment by ghost** [comment]
> Abductors do not need a "shrink ray" they arent meant to fuck with people on station

It isn't lethal and it's effect is temporary, it shouldn't be a problem

**Review by nesisknesis — APPROVED** [review]

**Comment by JamieD1** [comment]
Nope sorry.
